### PR TITLE
ui: added inspiration banner

### LINKED ISSUE #82: ui: inspiration banner

### LINKED ISSUE #85: curate: inspiration quote

### LINKED ISSUE #92: curate: images links

### PR DESCRIPTION
**Pull request by mayramtv**
Resolves #92 
resolves #85 
resolves #82 

ui: added inspiration banner; curated images and quotes: quotes are picked based on the day, not the mood.

@lenabaidakova or @claytonchew  could you please review this PR?

### CODE REVIEW

File: components/Banner.vue [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    </div>
+</template>
+
+
+<script setup>
+    import { _backgroundImage } from "#tailwind-config/theme";
+    import bannerImages from "../static/images.json" 
+    import bannerQuotes from "../static/quotes.json" 
+
+    const day = new Date().getDate();
```
[the remark this answers is not in the document]
Comment (mayramtv) [reply components/Banner.vue:34]: @lenabaidakova Yes, it currently changes only every day. But it doesn't changes when we switch journal day. Just to make sure I understand, does the image needs to change every day and every time we change journals?

**Review by lenabaidakova — APPROVED** [review]

File: components/Banner.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (mayramtv) [reply components/Banner.vue]: @lenabaidakova Oh sorry, I didn't see your message until a fixed it. I modify the code to pick the day based on the day selected for the journal or on the calendar. It does look much better and makes more sense. Thank you

**Comment by mayramtv** [comment]
@claytonchew I change the contrast of the text. Let me know if if it works better.

**Comment by claytonchew** [comment]
@mayramtv

I've pushed a few improvements, most are chores on formatting and typo, but one notable change commit is [refactor: separate json and logic handling from server](https://github.com/uol-student-life/student-life/pull/93/commits/068846395980f73e2ddb2bfac1b3ae90bc633f8b) in this PR, which I'll explain below why this is a better design pattern.

Since json is just a container of data, and any container of data we can treat it as some sort of a database. To ensure that your UI is compatible for any future change on how it presents this data, we can wrap the json and its logical handling on the server-side.

This essentially creates a clear **separation of concern** between the UI and the server, whereby 

- the server is responsible for fetching and handling data for the UI.
- the UI is responsible on how the fetched data is presented.

Hence, you can create an API that does the fetching and handling for the UI like so:

```typescript
// random.get.ts

import images from "~/static/images.json";

export default defineEventHandler(async (event) => {
  return images[Math.floor(Math.random() * images.length)];
});
```

Then, on the UI, it only needs to call this api to fetch (in this case) a random banner, like so:

```vue
<script setup>
const banner = ref(null);

const getImg = async () => {
  $fetch("/api/banner/random").then((response) => {
    banner.value = response;
  });
};

// this ensures that getImg() is called when page is mounted (loads)
onMounted(() => {
  getImg();
  getQuote();
});

// this ensures that getImg() is called again when current journal is changed
watch(
  () => props.currentJournal,
  () => {
    getImg();
  }
);
</script>

<template>
<div
    v-bind:style="{ backgroundImage: 'url(' + banner?.link + ')' }"
  >
 ...
</div>
</template>
```

This pattern allows future changes to occur on the server-side – it being how or where to retrieve data from. For example, you might want to integrate external image library sources and have the server to fetch those and return to the UI in the exact data model that the UI would consume.

Or it can also allow future changes to occur on the UI side – it being how it presents or consume the API.

Overall, you did a very good job with this PR with little to no bugs. Thanks @mayramtv

**Comment by mayramtv** [comment]
@claytonchew sorry I didn't answer before. I really appreciate the information and the explanation. I can understand better how it works. Thanks again.

**Pull request merged**